crates/pop-cli/src/main.rs

```diff
@@ -40,39 +45,115 @@ enum Commands {
 
 #[tokio::main]
 async fn main() -> Result<()> {
+	env_logger::init();
+	init_config()?;
+	// environment variable `POP_TELEMETRY_ENDPOINT` is evaluated at compile time
+	let endpoint =
+		option_env!("POP_TELEMETRY_ENDPOINT").unwrap_or("http://127.0.0.1:3000/api/send");
+
+	let maybe_config_file = config_file_path();
+	// if config file errors set telemetry to None, otherwise Some(tel)
+	let maybe_tel = maybe_config_file.ok().map(|path| Telemetry::new(endpoint.to_string(), path));
+
+	// handle for await not used here as telemetry should complete before any of the commands do.
+	// Sends a generic ping saying the CLI was used
+	spawn(record_cli_used(maybe_tel.clone()));
+
+	// type to represent static telemetry data. I.e., does not contain data dynamically chosen by user
+	// like in pop new parachain.
+	let mut tel_data: (&str, &str, Value) = ("", "", json!(""));
```

Comment:
As the telemetry data is only being written after command execution, it would be much simpler to just return tel_data from each of the match arms. The success type of the result is currently (), therefore unused and can therefore be easily replaced with tel data. Using a return type ensures that every condition returns telemetry, whereas this current approach could result in some condition simply not updating the telemetry data.